syntaxes/haskell.YAML-tmLanguage

```diff
@@ -573,6 +547,35 @@ repository:
           | }             # A block closed? Maybe this should also include `;`, because non-indentation based `do`
         )
         | ^(?!\1\s+\S|\s*$)    # at least one, no-further indented, non-whitespace character. I.e. a same-level declaration/implementation
+  type_application:
+    patterns:
+      - begin: '(?<=\s)(@)(\()'
```

Comment:
I am not sure about the look-behind here. As your original issue (#73) quoted, the `@` may not be preceeded by any end-of-identifier character, which is why I just copied the legal identifier character class regex. It may be that the majority there overlaps with the inversion of `\s`, but at least in cases where you have `(my expr)@type` it should probably still parse as a type application, what do you think?

Reply:
I realize this is a very niche case though :smile:

Reply:
Good point. It seems a bit subtle indeed, as actually

```haskell
f = (id)@([Int])
```
is not parsed by GHC, whereas

```haskell
f = [1,2,3]@[Int]
g = A{f=1}@Bool
```

are both parsed as type applications. I'm not sure exactly then which are the relevant characters, do you have a suggestion for a regex? Like `[\s\}\]]` or something.